module/module.go

```diff
@@ -772,10 +735,35 @@ func (m *Module) AddStream(ctx context.Context, req *streampb.AddStreamRequest)
 	m.logger.CInfow(ctx, "AddStream calling AddTrack", "name", name, "tlsRTP", tlsRTP)
 	sender, err := m.pc.AddTrack(tlsRTP)
 	if err != nil {
-		return nil, errors.Wrap(err, "error adding track")
+		err = errors.Wrap(err, "error adding track")
+		if unsubErr := vcss.Unsubscribe(ctx, sub.ID); unsubErr != nil {
+			return nil, multierr.Combine(err, unsubErr)
+		}
+		return nil, err
 	}
 
-	m.activeResourceStreams[name] = peerResourceState{sender: sender, subID: subID}
+	removeTrackOnSubTerminate := func() {
+		// wait until either the module is shutting down, or the subscription terminates
+		var msg string
+		select {
+		case <-sub.Context.Done():
+			msg = "rtp_passthrough subscription expired, calling RemoveTrack"
+		case <-m.ctx.Done():
+			msg = "module closing calling RemoveTrack"
+		}
+		// remove the track from the peer connection so that viam-server clients know that the stream has terminated
+		m.mu.Lock()
+		defer m.mu.Unlock()
+		m.logger.Debugw(msg, "name", name, "subID", sub.ID.String())
+		delete(m.activeResourceStreams, name)
+		if err := m.pc.RemoveTrack(sender); err != nil {
+			m.logger.Warnf("RemoveTrack returned error", "name", name, "subID", sub.ID.String(), "err", err)
+		}
+	}
```

Comment:
This allows the model's `Close` function, as well as module lifecycle events (reconfiguration, close, etc) to trigger cleaning up all WebRTC resources of subscriptions that have terminated.